src/java/grl/src/grl/driver/GRL_Driver.java

```diff
@@ -232,7 +262,9 @@ public void run()
 		while (!stop && !_startStopUI.is_stopped() && _lbr.getSafetyState().getEmergencyStopInt()==EmergencyStop.INACTIVE) {
 			message_counter+=1;
 			_currentKUKAiiwaState = udpMan.waitForNextMessage();
-			_previousKUKAiiwaState = udpMan.getPrevMessage();
+			
+			//Some bug in this, just set _previousKUKAiiwaState = _currentKUKAiiwaState before looping/continue?
+			_previousKUKAiiwaState = udpMan.getPrevMessage();  
```

Comment:
cool! then you should be able to make use of teach mode. It hasn't been tested in a while because I no longer have a robot that supports it. 

Also, could you describe the bug you encountered?